cmd/contour/serve.go

```diff
@@ -397,14 +397,24 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 			ResponseHeadersPolicy: &responseHeadersPolicy,
 		},
 	}
+
+	// Log that we're using the fallback certificate if configured.
+	if fallbackCert != nil {
+		log.WithField("context", "fallback-certificate").Infof("enabled fallback certificate with secret: %q", fallbackCert)
+	}
+
+	if clientCert != nil {
+		log.WithField("context", "envoy-client-certificate").Infof("enabled client certificate with secret: %q", clientCert)
+	}
+
```

Comment:
This seems like a win, but not really related to this change?

Reply:
This wasn't added, just moved around.

Reply:
We really need to refactor this whole section up. I think @skriss has an open issue for that. There are assumed ordering patterns that need to be kept in place making it difficult to manage.

Reply:
Yeah, I would definitely like to spend some time on this soon, this whole thing keeps getting more complicated and potentially error-prone. See #3453 for tracking.